bin/mantle-dynamics-L0/3_convection_in_box_model.ipynb

```diff
@@ -0,0 +1 @@
or x in nodes])\n","    values   = np.array([t for t in field])[:,0]\n","    fn       = interpolate.LinearNDInterpolator(coords,values)\n","    viz_data = (fn(mesh_x, mesh_y))\n","\n","    return viz_data"]},{"cell_type":"code","execution_count":null,"id":"1e838593","metadata":{"id":"1e838593"},"outputs":[],"source":["# Create figure, axis, and colorbar for the animation\n","fig, ax1 = plt.subplots(1, 1, figsize=(8, 4))\n","data_to_plot = read_solution_files(viz_files[0], 'T')\n","mesh     = ax1.pcolormesh(mesh_x, mesh_y, data_to_plot, cmap='RdBu_r')\n","colorbar = fig.colorbar(mesh, ax=ax1)\n","#---------------------------------------------------------------------------\n","\n","def animate(time_step):\n","    '''\n","    This function takes the time step, i, and plots the output solution field\n","    on the mesh.\n","\n","    Inputs:\n","    time_step (int)       : The time step number in the solution\n","    skip_time_steps (int) : Number of time steps between each graphical output\n","    '''\n","\n","    ax1.clear()\n","    data_to_plot = read_solution_files(viz_files[time_step], 'T')\n","    mesh = ax1.pcolormesh(mesh_x, mesh_y, data_to_plot, cmap='RdBu_r')\n","    colorbar.update_normal(mesh)\n","\n","    # Add a contour to approximate the shape of the convection cell\n","    ax1.contour(mesh_x, mesh_y, data_to_plot, levels=[1673], colors='k')\n","\n","    ax1.set_title(f'Temperature field at time step {time_step}')\n","    ax1.set_xlabel('X Coordinate')\n","    ax1.set_ylabel('Y Coordinate')\n","\n","    return [mesh]"]},{"cell_type":"code","execution_count":null,"id":"dc60407d","metadata":{"id":"dc60407d"},"outputs":[],"source":["# Now, let's plot the evolution of the temperatures in the box\n","\n","ani = animation.FuncAnimation(fig, animate, frames=(len(viz_files)))\n","HTML(ani.to_jshtml())"]},{"cell_type":"markdown","id":"a8384add","metadata":{"id":"a8384add"},"source":["#### Model Analysis\n","In the above cell, the black contour approximately delineates the convection cell. To the dominance of convection over conduction, we need to look at the Rayleigh number and Nusselt number."]},{"cell_type":"markdown","id":"9d39933e","metadata":{"id":"9d39933e"},"source":[" &nbsp;<div style=\"text-align: right\">  \n","    &rarr; <b>NEXT: [Rayleigh Nusselt Relationship](./4_Rayleigh_Nusselt_Relationship.ipynb) </b> <a href=\"\"></a> &nbsp;&nbsp;\n","     <img src=\"../assets/education-gem-notebooks_icon.png\" alt=\"icon\"  style=\"width:4%\">\n","  </div>"]},{"cell_type":"code","source":[],"metadata":{"id":"Y-psg6dlpjLL"},"id":"Y-psg6dlpjLL","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"display_name":"Python 3 (ipykernel)","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.11.5"},"colab":{"provenance":[]}},"nbformat":4,"nbformat_minor":5}
```

Comment:
<p>The last sentence seems awkward.  Suggested modification</p><p> </p><p>... we need to look at the Rayleigh and Nusselt numbers.</p><p>... we need to look at the Rayleigh Nusselt number relationship.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>